tests/functional/agent/agent_functional_test.py

```diff
@@ -327,6 +328,7 @@ def test_specific_model_parameters_e2e(tool_config):
         tools=[tool],
         llm_id="6626a3a8c8f1d089790cf5a2",  # Using LLM ID from test data
     )
+    agent.deploy()
```

Comment:
Why do we have to deploy?

Reply:
Without deploying, the second test is failing. The agent is not being overwritten, so it uses the search instead of the translation tool. Is that expected?